### PR TITLE
Richer context in the forge promote walk

### PR DESCRIPTION
**Pull request by amc-corey-cox**
The `forge promote` walk previously printed only the title before each y/N prompt, which made it hard to tell what you were promoting — and let a stale CI-validation PR get promoted by accident.

Each item now shows enough to decide without leaving the terminal:
- head -> base branch and opened date (for PRs; issues show the date)
- a first-paragraph excerpt of the body (skips markdown headings like `## Summary`, collapses whitespace, caps length)
- the Forgejo URL to click through for full detail

`list_promotable` now carries `body`, `url`, `created_at`, and `head`/`base`; a new `_promotable_summary` renders the block. All fields are read defensively so a minimal item still renders just its title. Adds unit tests for the excerpt and summary helpers.

### CODE REVIEW

File: src/cc_forge/promote.py [hunk 1]
```diff
@@ -209,11 +210,63 @@ def list_promotable(config: ForgeConfig, repo_name: str) -> list[dict]:
...
 
+def _first_paragraph(body: str, limit: int = 200) -> str:
+    """First real text paragraph of a body: skip markdown headings and blanks,
+    collapse whitespace, and cap the length."""
+    for para in body.strip().split("\n\n"):
+        text = " ".join(para.split())
+        if not text or text.startswith("#"):
+            continue
+        return text if len(text) <= limit else text[: limit - 3].rstrip() + "..."
+    return ""
```
[the remark this answers is not in the document]
Comment (amc-corey-cox) [reply src/cc_forge/promote.py]: Fixed. `_first_paragraph` now extracts line-by-line — skips leading headings/blanks and collects until the next blank line or heading — so a `## Summary` directly above text (no blank line) still yields the text. Also clamped the truncation so `limit < 3` can no longer exceed the limit.

File: tests/unit/test_promote.py [hunk 2]
```diff
@@ -345,6 +345,33 @@ def test_list_promotable_merges_issues_and_prs_sorted(monkeypatch):
     assert [(i["kind"], i["number"]) for i in items] == [("issue", 3), ("pr", 7)]
 
 
+def test_first_paragraph_skips_headings_and_caps():
+    body = "## Summary\n\nExpands the shim from 4 to 8 commands.\n\nMore detail."
+    assert promote_mod._first_paragraph(body) == "Expands the shim from 4 to 8 commands."
+    capped = promote_mod._first_paragraph("x" * 250, limit=50)
+    assert len(capped) == 50 and capped.endswith("...")
+    assert promote_mod._first_paragraph("") == ""
```
[the remark this answers is not in the document]
Comment (amc-corey-cox) [reply tests/unit/test_promote.py:353]: Added. The test now covers the no-blank-line pattern (`## Summary\nExpands…`) plus a `limit < 3` edge case.

File: src/cc_forge/promote.py [hunk 3]
```diff
@@ -209,11 +210,72 @@ def list_promotable(config: ForgeConfig, repo_name: str) -> list[dict]:
...
+    blanks, collect until the next blank line or heading, collapse whitespace,
+    and cap the length (even a heading directly above text, no blank line)."""
+    picked: list[str] = []
+    for line in body.strip().splitlines():
+        stripped = line.strip()
+        if not stripped or stripped.startswith("#"):
+            if picked:
+                break  # a blank line or heading ends the first paragraph
+            continue   # still skipping leading blanks/headings
+        picked.append(stripped)
```
[the remark this answers is not in the document]
Comment (amc-corey-cox) [reply src/cc_forge/promote.py:256]: Fixed. `_first_paragraph` now detects real ATX headings via `#{1,6}(?:\s|$)` instead of `startswith("#")`, so `#1 ...` refs and numbered lists are kept while `## Summary` is still skipped. Added a `#1` test case.

File: src/cc_forge/promote.py [hunk 4]
```diff
@@ -209,11 +210,77 @@ def list_promotable(config: ForgeConfig, repo_name: str) -> list[dict]:
...
+    first-paragraph excerpt, and the Forgejo URL."""
+    lines = [f"{it['kind'].upper()} #{it['number']}: {it['title']}"]
+    meta = []
+    if it.get("head") and it.get("base"):
+        meta.append(f"{it['head']} -> {it['base']}")
+    if it.get("created_at"):
+        meta.append(f"opened {it['created_at'][:10]}")
+    if meta:
+        lines.append("   " + " | ".join(meta))
+    excerpt = _first_paragraph(it.get("body", ""))
```
[the remark this answers is not in the document]
Comment (amc-corey-cox) [reply src/cc_forge/promote.py]: Fixed. `_promotable_summary` now uses `it.get("body") or ""`, so an explicit `None` body is coerced to empty instead of crashing `_first_paragraph`. Added a `{"body": None}` test.
